crates/uv-pep508/src/marker/tree.rs

```diff
@@ -2528,6 +2537,76 @@ mod test {
         );
     }
 
+    #[test]
+    fn test_python_version_equal_star() {
+        // Input, equal python_version, equal, python_full_version
+        let cases = [
+            ("3.*", "3.*", "3.*"),
+            ("3.0.*", "3.0", "3.0.*"),
+            ("3.0.0.*", "3.0", "3.0.*"),
+            ("3.9.*", "3.9", "3.9.*"),
+            ("3.9.0.*", "3.9", "3.9.*"),
+            ("3.9.0.0.*", "3.9", "3.9.*"),
+        ];
+
+        for (input, equal_python_version, equal_python_full_version) in cases {
+            assert_eq!(
+                m(&format!("python_version == '{input}'")),
+                m(&format!("python_version == '{equal_python_version}'")),
+                "{input} {equal_python_version}"
+            );
+            assert_eq!(
+                m(&format!("python_version == '{input}'")),
+                m(&format!(
+                    "python_full_version == '{equal_python_full_version}'"
+                )),
+                "{input} {equal_python_full_version}"
+            );
+        }
+
+        let cases_false = ["3.9.1.*", "3.9.1.0.*", "3.9.1.0.0.*"];
+
+        for input in cases_false {
+            assert!(
+                m(&format!("python_version == '{input}'")).is_false(),
+                "{input}"
+            );
+        }
+    }
+
+    #[test]
+    fn test_tilde_equal_normalization() {
```

Comment:
This does not yet include the `test_tilde_equal_inverted_normalization` test from #14259.